File: lib/models/saved_project.dart

class SavedProject {
  final String id;
  final String name;
  final String description;
  final List<String> tags;
  final DateTime lastModified;
  final String jsonContent; // The full project export JSON

  SavedProject({
    required this.id,
    required this.name,
    required this.description,
    required this.tags,
    required this.lastModified,
    required this.jsonContent,
  });

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'description': description,
        'tags': tags,
        'lastModified': lastModified.toIso8601String(),
        'jsonContent': jsonContent,
      };

  factory SavedProject.fromJson(Map<String, dynamic> json) {
    return SavedProject(
      id: json['id'],
      name: json['name'],
      description: json['description'] ?? '',
      tags: List<String>.from(json['tags'] ?? []),
      lastModified: DateTime.parse(json['lastModified']),
      jsonContent: json['jsonContent'],
    );
  }
}

File: src/models/SavedProject.js
// Converted from Dart SavedProject model
export class SavedProject {
  constructor({ id, name, description = '', tags = [], lastModified, jsonContent }) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.tags = tags;
    this.lastModified = lastModified ? new Date(lastModified) : new Date();
    this.jsonContent = jsonContent;
  }

  toJson() {
    return {
      id: this.id,
      name: this.name,
      description: this.description,
      tags: this.tags,
      lastModified: this.lastModified.toISOString(),
      jsonContent: this.jsonContent,
    };
  }

  static fromJson(json) {
    return new SavedProject({
      id: json.id,
      name: json.name,
      description: json.description || '',
      tags: json.tags || [],
      lastModified: json.lastModified,
      jsonContent: json.jsonContent,
    });
  }
}
